src/utils/properties.js

```diff
@@ -90,59 +85,79 @@ const propSort = (a, b) => {
 	}
 };
 
-export const renderStaticProperties = (properties, isHoc) => {
+export const renderStaticProperties = async (properties, isHoc) => {
 	if (!properties.static.length) {
 		return;
 	}
-	properties.static = properties.static.sort(propSort);
+
+	// create an array with the required static properties
+	const isRequiredTag = await Promise.all(properties.static.map(async (prop) => await hasRequiredTag(prop)));
+	// get all the required static properties and sort them. after that sort all the non-required properties and concat them to sorted required properties
+	properties.static = properties.static.filter((el, index) => isRequiredTag[index]).sort(propSort).concat(properties.static.filter((el, index) => !isRequiredTag[index]).sort(propSort));
 	if (isHoc) {
-		return renderHocConfig(properties.static[0]);
+		return await renderHocConfig(properties.static[0]);
 	} else {
 		return (
 			<section className={css.statics}>
 				{properties.static.length ? <h5>Statics</h5> : null}
 				<dl>
-					{properties.static.map(renderProperty)}
+					{await Promise.all(properties.static.map(renderProperty))}
 				</dl>
 			</section>
 		);
 	}
 };
 
-export const renderInstanceProperties = (properties, isHoc) => {
+export const renderInstanceProperties = async (properties, isHoc) => {
 	if (!properties.instance.length) {
 		return;
 	}
-	const instanceProps = properties.instance.filter(prop => prop.kind !== 'function').sort(propSort);
-	const instanceMethods = properties.instance.filter(prop => prop.kind === 'function').sort(propSort);
+
+	// create an array with the required properties
+	const hasRequiredTagMethods = await Promise.all(properties.instance.map(async (prop) => await hasRequiredTag(prop) && prop.kind === 'function'));
+	// get all the properties that have the required tag and sort them
+	let instanceMethods = properties.instance.filter((el, index) => hasRequiredTagMethods[index] && el.kind === 'function').sort(propSort);
+	// sort all non-required properties and concat them to sorted required properties
+	instanceMethods = instanceMethods.concat(properties.instance.filter((el, index) => !hasRequiredTagMethods[index] && el.kind === 'function').sort(propSort));
+
+	// create an array with the required methods
+	const hasRequiredTagProps = await Promise.all(properties.instance.map(async (prop) => await hasRequiredTag(prop) && prop.kind !== 'function'));
+	// get all the methods that have the required tag and sort them
+	let instanceProps = properties.instance.filter((el, index) => hasRequiredTagProps[index] && el.kind !== 'function').sort(propSort);
+	// sort all non-required methods and concat them to sorted required methods
+	instanceProps = instanceProps.concat(properties.instance.filter((el, index) => !hasRequiredTagProps[index] && el.kind !== 'function').sort(propSort));
+
```

Comment:
Note: I had to first create an array with the required properties/methods, sort them alphabetically, then get all the non-required properties/methods, sort them alphabetically, then
add them to the sorted requred properties/methods.